### DISCUSSION TITLE
Problem json encoding AggregationResult

### DISCUSSION THREAD

**Discussion by danielcmm** [Jeroen-G/Explorer#87] [General]
Hi!

When we try to return the aggregations from a search result, the library returns a list of AggregationResult objects.
This object has no public properties, so, when laravel tries to json_encode them, nothing is returned.

One simple solution would be to make the attributes public. What u guys think?

Thanks!

**Comment by Jeroen-G** [comment]
> when laravel tries to json_encode them

When does that happen?

**Reply by danielcmm** [reply]
When you try to return it from a Controller, for example.

`return [
'aggs' => $results->aggregations()
]`

**Reply by Jeroen-G** [reply]
I think what I would do is create a viewmodel/DTO/response object that can be serialised and fill it's properties based on the search results. That way the output of the application is less dependent on a third party package. Does that make some sense to you?

I am very sure I am not going to make those aggregation properties public and, at least for now, I am holding off on making that class serialisable.

**Reply by danielcmm** [reply]
I think its an overkill but ok... i mean, the objects are all set and ready to be returned, didn´t want to loop and make other objects just to make the properties availiable. Anyway, i understand your point. 

Really appreciate your support! cheers
